frame/treasury/src/lib.rs

```diff
@@ -221,6 +269,25 @@ decl_storage! {
 		/// Simple preimage lookup from the reason's hash to the original data. Again, has an
 		/// insecure enumerable hash since the key is guaranteed to be the result of a secure hash.
 		pub Reasons get(fn reasons): map hasher(identity) T::Hash => Option<Vec<u8>>;
+
+		/// Number of bounty proposals that have been made.
+		pub BountyCount get(fn bounty_count): BountyIndex;
+
+		/// Bounties that have been made.
```

Comment:
It's bad form to have three separate map items (four if you count `BountyApprovals`, which should probably also be associated with `BountyIndex`) since it will place 3x the number of items in the trie needlessly. Unless each item contains unconstrained encode/decode complexity (they don't here), then just place them all in one item.

If you ever see multiple maps in the same pallet with the same key type, it's a red flag.

Reply:
Currently `struct Bounty` is immutable and contains description which can be large (won't be an issue any more if type changed to `Data`).

Should I move description out to own storage similar to `Reasons` and have `BountyStatuses` and `BountyBeneficiary` merged into `Bounties`?

Reply:
Yeah I think so.